pandas/core/arrays/base.py

```diff
@@ -610,3 +618,121 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+class ExtensionOpsBase(object):
+    """
+    A base class for the mixins for different operators.
+    Can also be used to define an individual method for a specific
+    operator using the class method create_method()
+    """
+    @classmethod
+    def create_method(cls, op):
+        """
+        A class method that returns a method that will correspond to an
+        operator for an ExtensionArray subclass.
+
+        Parameters
+        ----------
+        op:   An operator that takes arguments op(a, b)
+
+        Returns
+        -------
+        A method that can be bound to a method of a class
+
+        Usage
+        -----
+        Given an ExtensionArray subclass called MyClass, use
+
+        mymethod = create_method(my_operator)
+        in the class definition of MyClass to create the operator
+
+        """
+        op_name = ops._get_op_name(op, False)
+
+        def _binop(self, other):
+            def convert_values(parm):
+                if isinstance(parm, ExtensionArray):
+                    ovalues = list(parm)
+                elif is_extension_array_dtype(parm):
+                    ovalues = parm.values
+                elif is_list_like(parm):
+                    ovalues = parm
+                else:  # Assume its an object
+                    ovalues = [parm] * len(self)
+                return ovalues
+            lvalues = convert_values(self)
+            rvalues = convert_values(other)
+
+            try:
+                res = [op(a, b) for (a, b) in zip(lvalues, rvalues)]
+            except TypeError:
+                msg = ("ExtensionDtype invalid operation " +
+                       "{opn} between {one} and {two}")
+                raise TypeError(msg.format(opn=op_name,
+                                           one=type(lvalues),
+                                           two=type(rvalues)))
+
+            res_values = com._values_from_object(res)
+
+            try:
+                res_values = self._from_sequence(res_values)
+            except TypeError:
```

Comment:
would it makes sense to have a separate `_make_comparision_op` and `_make_aithmetics_op` ? 
The the comparison one should not need to try the conversion to ExtensionArray (if you want this to happen like in your case, you can still use the arithmetic one to add the comparison ops manually on your class)

Reply:
I see your point.  But if someone else had the same use case as me, where the comparison ops need to return an object rather than a boolean, then they'd have to know the workaround you suggest, and so then the question is whether we document it.

I'm leaving this as is for now, but will accept if you really want me to change it.

Reply:
One reason to split it up would be to avoid the extra overhead to try to convert it to an ExtensionArray in case of the boolean ops (although for a good implementation this probably should not be too much overhead ..)

Reply:
OK, I've done the split for the next commit.  I added a parameter to `create_method()` to specify whether to try the coercion to the underlying dtype, and said to not do it for the comparison operators.